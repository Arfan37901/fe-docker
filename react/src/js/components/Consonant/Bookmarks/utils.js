// eslint-disable-next-line import/prefer-default-export
export const selector = ({
    bookmarks: {
        leftFilterPanel: { selectBookmarksIcon, unselectBookmarksIcon },
        i18n: {
            leftFilterPanel: { filterTitle },
        },
    },
}) => ({
    label: filterTitle,
    selectIcon: selectBookmarksIcon,
    unselectIcon: unselectBookmarksIcon,
});
