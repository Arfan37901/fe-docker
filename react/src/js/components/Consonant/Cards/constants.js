
const defaultProps = {
    id: '',
    title: '',
    label: '',
    footer: '',
    ctaLink: '',
    endTime: '',
    logoAlt: '',
    logoSrc: '',
    videoURL: '',
    badgeText: '',
    startTime: '',
    cardImage: '',
    bannerIcon: '',
    description: '',
    logoBorderColor: '',
    bannerFontColor: '',
    isBookmarked: false,
    bannerDescription: '',
    disableBookmarkIco: '',
    logoBackgroundColor: '',
    bannerBackgroundColor: '',
};

export const SquareCardDefaultProps = {
    ...defaultProps,
    ctaLink: '',
};

export const FullCardDefaultProps = {
    ...defaultProps,
    ctaLink: '',
};

export const WideCardDefaultProps = {
    ...defaultProps,
    footer: [],
    isBookmarked: false,
    disableBookmarkIco: false,
};
