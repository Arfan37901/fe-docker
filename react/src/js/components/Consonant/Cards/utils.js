// eslint-disable-next-line import/prefer-default-export
export const selector = ({
    language,
    collection: {
        i18n: { prettyDateIntervalFormat },
    },
}) => ({
    locale: language,
    dateTemlate: prettyDateIntervalFormat,
});
