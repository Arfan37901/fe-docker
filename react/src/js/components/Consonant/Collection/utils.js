export const selector = ({
    pagination: { type },
    collection: { cardStyle },
}) => ({
    cardStyle,
    paginationType: type,
});

export const getFlatternProps = ({
    id,
    footer,
    ctaLink,
    isBookmarked,
    disableBookmarkIco,
    styles: {
        backgroundImage: cardImage,
    },
    contentArea: {
        title,
        description,
        detailText: label,
        dateDetailText: {
            endTime,
            startTime,
        },
    },
    overlays: {
        banner: {
            icon: bannerIcon,
            fontColor: bannerFontColor,
            description: bannerDescription,
            backgroundColor: bannerBackgroundColor,
        },
        videoButton: {
            url: videoURL,
        },
        logo: {
            src: logoSrc,
            alt: logoAlt,
            borderColor: logoBorderColor,
            backgroundColor: logoBackgroundColor,
        },
        label: {
            description: badgeText,
        },
    },
}) => ({
    id,
    title,
    label,
    footer,
    ctaLink,
    endTime,
    logoAlt,
    logoSrc,
    videoURL,
    badgeText,
    startTime,
    cardImage,
    bannerIcon,
    description,
    isBookmarked,
    logoBorderColor,
    bannerFontColor,
    bannerDescription,
    disableBookmarkIco,
    logoBackgroundColor,
    bannerBackgroundColor,
});
