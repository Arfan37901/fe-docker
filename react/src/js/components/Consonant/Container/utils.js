// eslint-disable-next-line import/prefer-default-export
export const selector = ({
    featuredCards,
    sort: {
        defaultSort,
        options: sortOptions,
    },
    bookmarks: {
        leftFilterPanel: { bookmarkOnlyCollection: onlyShowBookmarks },
    },
    collection: {
        resultsPerPage,
        endpoint: collectionEndpoint,
        totalCardsToShow: totalCardLimit,
        i18n: {
            onErrorTitle: apiFailureTitle,
            onErrorDescription: apiFailureDescription,
        },
    },
    search: {
        searchFields,
        noResultsTitle,
        noResultsDescription,
        i18n: {
            filterInfo: { searchPlaceholderText },
            topFilterPanel: { searchPlaceholderText: topPanelSearchPlaceholder },
            leftFilterPanel: { searchPlaceholderText: leftPanelSearchPlaceholder },
        },
    },
    pagination: {
        type: paginationType,
        enabled: paginationIsEnabled,
    },
    filterPanel: {
        filterLogic,
        type: filterPanelType,
        filters: authoredFilters,
        enabled: filterPanelEnabled,
    },
}) => ({
    defaultSort,
    sortOptions,
    filterLogic,
    searchFields,
    featuredCards,
    paginationType,
    noResultsTitle,
    resultsPerPage,
    totalCardLimit,
    apiFailureTitle,
    authoredFilters,
    filterPanelType,
    onlyShowBookmarks,
    collectionEndpoint,
    filterPanelEnabled,
    paginationIsEnabled,
    noResultsDescription,
    apiFailureDescription,
    searchPlaceholderText,
    topPanelSearchPlaceholder,
    leftPanelSearchPlaceholder,
});
