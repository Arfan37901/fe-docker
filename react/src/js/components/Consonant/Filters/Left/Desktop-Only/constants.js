export const selectedItemsDefaultProps = {
    numItemsSelected: 0,
};

export const clearButtonDefaultProps = {
    clearAllFiltersText: '',
};

export const titleDefaultProps = {
    panelHeader: '',
};
