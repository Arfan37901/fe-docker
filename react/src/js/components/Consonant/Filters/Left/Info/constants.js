
/**
 * Minimal viewport width to fit desktops/laptops
 * @type {Number}
 */
export const DESKTOP_MIN_WIDTH = 1200;

export const defaultProps = {
    cardsQty: 0,
    filtersQty: 0,
    selectedFiltersQty: 0,
    windowWidth: window.innerWidth,
};
