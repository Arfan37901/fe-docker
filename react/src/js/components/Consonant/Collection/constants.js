import { DEFAULT_SHOW_ITEMS_PER_PAGE } from '../Helpers/constants';

// eslint-disable-next-line import/prefer-default-export
export const defaultProps = {
    pages: 1,
    cards: [],
    resultsPerPage: DEFAULT_SHOW_ITEMS_PER_PAGE,
};
