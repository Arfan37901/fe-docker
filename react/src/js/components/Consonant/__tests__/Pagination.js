import { screen, fireEvent, getNodeText } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Pagination from '../Pagination';

import { PAGE_LIST, PAGES_PROPS, DEFAULT_PROPS } from './constants/pagination';

import makeSetup from './utils/settings';
import { getItemsRange } from './utils/pagination';

const setup = makeSetup(Pagination, DEFAULT_PROPS);

describe('Consonant/Pagination', () => {
    test('should render different items range', () => {
        PAGE_LIST.forEach((page) => {
            const { props: { showItemsPerPage }, wrapper } = setup({ currentPageNumber: page });

            const itemRange = getItemsRange({ page, itemsPerPage: showItemsPerPage });

            const paginationSummaryElement = screen.getByTestId('pagination--summary');

            expect(paginationSummaryElement).toHaveTextContent(itemRange);
            wrapper.unmount();
        });
    });

    describe('Check snapshots', () => {
        test('should renders correctly pagination range', () => {
            PAGES_PROPS.forEach((props) => {
                const { tree } = setup(props);

                expect(tree).toMatchSnapshot();
            });
        });
    });

    describe('Interaction with UI', () => {
        test('should call onClick', () => {
            const { props: { onClick } } = setup();

            const prevButton = screen.getByTestId('btn_prev');
            const nextButton = screen.getByTestId('btn_next');
            const pageButton = screen.queryAllByTestId('btn_page')[0];

            fireEvent.click(nextButton);

            expect(onClick).toHaveBeenCalledWith(2);

            fireEvent.click(prevButton);

            expect(onClick).toHaveBeenCalledWith(1);

            fireEvent.click(pageButton);
            const buttonText = getNodeText(pageButton);

            expect(onClick).toHaveBeenCalledWith(Number(buttonText));
        });
        test('should call onClick with 1', () => {
            const { props: { onClick } } = setup({ currentPageNumber: 0 });

            const prevButton = screen.getByTestId('btn_prev');

            fireEvent.click(prevButton);

            expect(onClick).toHaveBeenCalledWith(1);
        });
        test('should call onClick with totalPages', () => {
            const { props: { onClick } } = setup({ currentPageNumber: 2, totalPages: 1 });

            const nextButton = screen.getByTestId('btn_next');

            fireEvent.click(nextButton);

            expect(onClick).toHaveBeenCalledWith(1);
        });
        test('should call onClick with currentPageNumber - 1', () => {
            const { props: { onClick } } = setup({ currentPageNumber: 5 });

            const prevButton = screen.getByTestId('btn_prev');

            fireEvent.click(prevButton);

            expect(onClick).toHaveBeenCalledWith(4);
        });
        test('should call onClick not with 0', () => {
            const { props: { onClick } } = setup({ currentPageNumber: 1 });

            const prevButton = screen.getByTestId('btn_prev');

            fireEvent.click(prevButton);

            expect(onClick).toHaveBeenCalledWith(1);
        });
    });
});
