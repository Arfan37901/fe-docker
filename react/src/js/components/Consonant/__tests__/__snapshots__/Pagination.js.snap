// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Pagination Check snapshots should renders correctly pagination range 1`] = `
<div
  className="consonant-pagination"
>
  <div
    className="consonant-pagination--paginator"
  >
    <button
      className="consonant-pagination--btn consonant-pagination--btn_prev"
      data-testid="btn_prev"
      onClick={[Function]}
      type="buttton"
    >
      previous
    </button>
    <ul
      className="consonant-pagination--items"
    >
      <li
        className="consonant-pagination--item"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </li>
    </ul>
    <button
      className="consonant-pagination--btn consonant-pagination--btn_next"
      data-testid="btn_next"
      onClick={[Function]}
      type="buttton"
    >
      next
    </button>
  </div>
  <div
    className="consonant-pagination--summary"
    data-testid="pagination--summary"
  >
    Showing
    <strong>
      -9-0
       of
    </strong>
    100
     Results
  </div>
</div>
`;

exports[`Consonant/Pagination Check snapshots should renders correctly pagination range 2`] = `
<div
  className="consonant-pagination"
>
  <div
    className="consonant-pagination--paginator"
  >
    <button
      className="consonant-pagination--btn consonant-pagination--btn_prev"
      data-testid="btn_prev"
      onClick={[Function]}
      type="buttton"
    >
      previous
    </button>
    <ul
      className="consonant-pagination--items"
    >
      <li
        className="consonant-pagination--item consonant-pagination--item_active"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </li>
      <li
        className="consonant-pagination--item"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
      </li>
    </ul>
    <button
      className="consonant-pagination--btn consonant-pagination--btn_next"
      data-testid="btn_next"
      onClick={[Function]}
      type="buttton"
    >
      next
    </button>
  </div>
  <div
    className="consonant-pagination--summary"
    data-testid="pagination--summary"
  >
    Showing
    <strong>
      1-10
       of
    </strong>
    100
     Results
  </div>
</div>
`;

exports[`Consonant/Pagination Check snapshots should renders correctly pagination range 3`] = `
<div
  className="consonant-pagination"
>
  <div
    className="consonant-pagination--paginator"
  >
    <button
      className="consonant-pagination--btn consonant-pagination--btn_prev"
      data-testid="btn_prev"
      onClick={[Function]}
      type="buttton"
    >
      previous
    </button>
    <ul
      className="consonant-pagination--items"
    >
      <li
        className="consonant-pagination--item"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </li>
      <li
        className="consonant-pagination--item consonant-pagination--item_active"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
      </li>
    </ul>
    <button
      className="consonant-pagination--btn consonant-pagination--btn_next"
      data-testid="btn_next"
      onClick={[Function]}
      type="buttton"
    >
      next
    </button>
  </div>
  <div
    className="consonant-pagination--summary"
    data-testid="pagination--summary"
  >
    Showing
    <strong>
      -9-0
       of
    </strong>
    100
     Results
  </div>
</div>
`;

exports[`Consonant/Pagination Check snapshots should renders correctly pagination range 4`] = `
<div
  className="consonant-pagination"
>
  <div
    className="consonant-pagination--paginator"
  >
    <button
      className="consonant-pagination--btn consonant-pagination--btn_prev"
      data-testid="btn_prev"
      onClick={[Function]}
      type="buttton"
    >
      previous
    </button>
    <ul
      className="consonant-pagination--items"
    >
      <li
        className="consonant-pagination--item consonant-pagination--item_active"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </li>
      <li
        className="consonant-pagination--item"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
      </li>
    </ul>
    <button
      className="consonant-pagination--btn consonant-pagination--btn_next"
      data-testid="btn_next"
      onClick={[Function]}
      type="buttton"
    >
      next
    </button>
  </div>
  <div
    className="consonant-pagination--summary"
    data-testid="pagination--summary"
  >
    Showing
    <strong>
      1-10
       of
    </strong>
    100
     Results
  </div>
</div>
`;

exports[`Consonant/Pagination Check snapshots should renders correctly pagination range 5`] = `
<div
  className="consonant-pagination"
>
  <div
    className="consonant-pagination--paginator"
  >
    <button
      className="consonant-pagination--btn consonant-pagination--btn_prev"
      data-testid="btn_prev"
      onClick={[Function]}
      type="buttton"
    >
      previous
    </button>
    <ul
      className="consonant-pagination--items"
    >
      <li
        className="consonant-pagination--item consonant-pagination--item_active"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </li>
    </ul>
    <button
      className="consonant-pagination--btn consonant-pagination--btn_next"
      data-testid="btn_next"
      onClick={[Function]}
      type="buttton"
    >
      next
    </button>
  </div>
  <div
    className="consonant-pagination--summary"
    data-testid="pagination--summary"
  >
    Showing
    <strong>
      1-10
       of
    </strong>
    100
     Results
  </div>
</div>
`;

exports[`Consonant/Pagination Check snapshots should renders correctly pagination range 6`] = `
<div
  className="consonant-pagination"
>
  <div
    className="consonant-pagination--paginator"
  >
    <button
      className="consonant-pagination--btn consonant-pagination--btn_prev"
      data-testid="btn_prev"
      onClick={[Function]}
      type="buttton"
    >
      previous
    </button>
    <ul
      className="consonant-pagination--items"
    >
      <li
        className="consonant-pagination--item"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </li>
    </ul>
    <button
      className="consonant-pagination--btn consonant-pagination--btn_next"
      data-testid="btn_next"
      onClick={[Function]}
      type="buttton"
    >
      next
    </button>
  </div>
  <div
    className="consonant-pagination--summary"
    data-testid="pagination--summary"
  >
    Showing
    <strong>
      -9-0
       of
    </strong>
    100
     Results
  </div>
</div>
`;
